module/zfs/spa.c

```diff
@@ -1647,7 +1647,13 @@ spa_load_l2cache(spa_t *spa)
 	VERIFY(nvlist_remove(sav->sav_config, ZPOOL_CONFIG_L2CACHE,
 	    DATA_TYPE_NVLIST_ARRAY) == 0);
 
-	l2cache = kmem_alloc(sav->sav_count * sizeof (void *), KM_SLEEP);
+	/*
+	 * -fsanitize in userspace will detect kmem_alloc(0, ...)
+	 * as a leak
+	 */
+	if (sav->sav_count > 0)
+		l2cache = kmem_alloc(sav->sav_count * sizeof (void *),
```

Comment:
It looks like this same issue can occur on line 1588, I'm surprised it wasn't also flagged.